src/backend/gl/src/conv.rs

```diff
@@ -34,3 +33,13 @@ pub fn wrap_to_gl(w: i::WrapMode) -> t::GLenum {
         i::WrapMode::Border => gl::CLAMP_TO_BORDER,
     }
 }
+
+pub fn buffer_usage_to_gl_target(usage: buffer::Usage) -> Option<t::GLenum> {
+    match usage & (buffer::CONSTANT | buffer::INDEX | buffer::VERTEX | buffer::INDIRECT) {
+        buffer::CONSTANT => Some(gl::UNIFORM_BUFFER),
+        buffer::INDEX => Some(gl::ELEMENT_ARRAY_BUFFER),
+        buffer::VERTEX => Some(gl::ARRAY_BUFFER),
+        buffer::INDIRECT => unimplemented!(),
+        _ => None
```

Comment:
Do we need to return `None` here? Feels a bit weird to return an error when we actually have a GL version which supports different targets but older versions would just use `ARRAY_BUFFER`.

Reply:
Some GL platforms (ES, looking at you) don't allow changing the bind target from what was bound for the first time. Returning `None` here has the semantics of: no target matches all the required `Usage`. The calling side then decides if it needs to throw up, or use a default target (`ARRAY_BUFFER`).